src/libsync/discoveryphase.cpp

```diff
@@ -100,156 +87,90 @@ bool DiscoveryJob::checkSelectiveSyncNewFolder(const QString &path, RemotePermis
         // no limit, everything is allowed;
-        return false;
-    }
-
-    // Go in the main thread to do a PROPFIND to know the size of this folder
-    qint64 result = -1;
-
-    {
-        QMutexLocker locker(&_vioMutex);
-        emit doGetSizeSignal(path, &result);
-        _vioWaitCondition.wait(&_vioMutex);
+        return callback(false);
     }
 
-    if (result >= limit) {
-        // we tell the UI there is a new folder
-        emit newBigFolder(path, false);
-        return true;
-    } else {
-        // it is not too big, put it in the white list (so we will not do more query for the children)
-        // and and do not block.
-        auto p = path;
-        if (!p.endsWith(QLatin1Char('/'))) {
-            p += QLatin1Char('/');
+    // do a PROPFIND to know the size of this folder
+    auto propfindJob = new PropfindJob(_account, _remoteFolder + path, this);
+    propfindJob->setProperties(QList<QByteArray>() << "resourcetype"
+                                                   << "http://owncloud.org/ns:size");
+    QObject::connect(propfindJob, &PropfindJob::finishedWithError,
+        this, [=] { return callback(false); });
+    QObject::connect(propfindJob, &PropfindJob::result, this, [=](const QVariantMap &values) {
+        auto result = values.value(QLatin1String("size")).toLongLong();
+        if (result >= limit) {
+            // we tell the UI there is a new folder
+            emit newBigFolder(path, false);
+            return callback(true);
+        } else {
+            // it is not too big, put it in the white list (so we will not do more query for the children)
+            // and and do not block.
+            auto p = path;
+            if (!p.endsWith(QLatin1Char('/')))
+                p += QLatin1Char('/');
+            _selectiveSyncWhiteList.insert(
+                std::upper_bound(_selectiveSyncWhiteList.begin(), _selectiveSyncWhiteList.end(), p),
+                p);
+            return callback(false);
         }
-        _selectiveSyncWhiteList.insert(std::upper_bound(_selectiveSyncWhiteList.begin(),
-                                           _selectiveSyncWhiteList.end(), p),
-            p);
-
-        return false;
-    }
+    });
+    propfindJob->start();
 }
 
-int DiscoveryJob::checkSelectiveSyncNewFolderCallback(void *data, const QByteArray &path, RemotePermissions remotePerm)
+/* Given a path on the remote, give the path as it is when the rename is done */
+QString DiscoveryPhase::adjustRenamedPath(const QString &original) const
 {
-    return static_cast<DiscoveryJob *>(data)->checkSelectiveSyncNewFolder(QString::fromUtf8(path), remotePerm);
+    int slashPos = original.size();
+    while ((slashPos = original.lastIndexOf('/', slashPos - 1)) > 0) {
+        auto it = _renamedItems.constFind(original.left(slashPos));
+        if (it != _renamedItems.constEnd()) {
+            return *it + original.mid(slashPos);
+        }
+    }
+    return original;
```

Comment:
We don't need them because we track the exact paths in the different part of `ProcessDirectoryJob::PathTuple`, so there is no need to look up things later.

(In theory, this should allow more complex renames to happen  (rename A/B/ to A/B2/ on the server, and A/B/C to A/B/C2 on the client) although i haven't test that yet.)